pkg/pool-weighted/contracts/BaseWeightedPool.sol

```diff
@@ -139,26 +139,26 @@ abstract contract BaseWeightedPool is BaseMinimalSwapInfoPool {
     function _beforeJoinExit(
         uint256[] memory, /* preBalances */
         uint256[] memory /* normalizedWeights */
-    ) internal virtual returns (uint256) {
-        return totalSupply();
+    ) internal virtual returns (uint256, uint256) {
```

Comment:
The comment says "Returns the Pool's total supply by default" - should also mention the second return value. Maybe something like:
```
 * @dev Called before any join or exit operation. Returns the Pool's total supply and zero due protocol fees
 * by default, but derived contracts may add custom behavior here to calculate a BPT protocol fee amount.
 * This often has to do with optimizing the collection of fees from multiple sources.
```

Reply:
I've added some `@return` tags to the natspec. Would you be happy if we removed the references to the default return values in here and just had something like:

```
    /**
     * @dev Called before any join or exit operation. Derived contracts may add custom behavior here to calculate any
     * fees to be paid to the protocol (or e.g. the Pool's owner) before the join or exit.
     * @return preJoinExitTotalSupply - the total supply once all fees prior to the join/exit have been paid.
     * @return preJoinExitProtocolFees - the amount of protocol fees which are to be collected before the join/exit.
     */
```

Reply:
Yes; I noted those were still there and a bit stale. That looks good.